core/trino-main/src/main/java/io/trino/execution/buffer/OutputBufferMemoryManager.java

```diff
@@ -119,6 +134,17 @@ public void updateMemoryUsage(long bytesAdded)
         }
     }
 
+    private synchronized void recordBufferUtilization()
+    {
+        long recordTime = ticker.read();
+        long timeDiff = recordTime - this.lastBufferUtilizationRecordTime;
+        if (timeDiff > 0) {
+            bufferUtilization.add(lastBufferUtilization, (double) timeDiff);
+            lastBufferUtilizationRecordTime = recordTime;
+            lastBufferUtilization = getUtilization();
```

Comment:
you still should record current `lastBufferUtilization` even if `timeDiff` is small